crates/sqlexec/src/planner/session_planner.rs

```diff
@@ -228,18 +226,21 @@ impl<'a> SessionPlanner<'a> {
                     other => return Err(internal!("Unknown catalog type: {}", other)),
                 };
 
+                let mut storage_options = StorageOptions::try_from(m)?;
+                if let Some(creds) = creds_options {
+                    storage_options_with_credentials(&mut storage_options, creds);
```

Comment:
shouldn't we return storage_options rather than pass the return by value? I might be new to the idiom, but this always looks harder?

also similar idiom/style question: do we want function names to look like verb-phrases? (e.g. get_storage..., or resolve_storage?)

Reply:
The goal here is to "hydrate" the supplied options with additional options extracted from the provided credentials object. The name is a bit cumbersome I admit; but still I feel it's better to update the options in place rather than create a new map and re-populate all the options again.

Reply:
yeah, that makes sense. I guess I'd be inclined to do that kind of thing in a method on the options rather than a function but it doesn't matter much.

Reply:
> I guess I'd be inclined to do that kind of thing in a method on the options

True, that is a better solution. However this comment
https://github.com/GlareDB/glaredb/blob/fdbf9118078a97dbc8836836f2f75f22ef5c9b55/crates/protogen/src/lib.rs#L5
dissuaded me from putting that there, even though (alongside with `storage_options_into_object_store`) it fits better there.

Reply:
You're probably more in touch with the idiom on this kind of thing, and there's often a very legitimate reason to avoid collecting "model" logic in the these packages that can't depend on anything else internally. 

At the same time, the thing that happens with a strict "don't put logic here" rule, is that you end up with all the logic in different places and that's bad too. I wonder if: 
- we can put logic here without dependencies on other package for small things
- we can import/re-export these models via other packages and put the logic in _those_ packages just for clarity?
- decide that it's fine as it is and that the confusion (potential) is ok for the medium term, and ignore.